### ISSUE TITLE
KeyError: 'greyBoxMessage' when checking fares (companion check)

### ISSUE THREAD

**Issue by Ohmic43** [jdholtz/auto-southwest-check-in#381]
### Version

Version 9.0

### Description

Ran the latest migrate-to-sw-web-api and faced this Fare Checker issue. 
This is with account login

### To Reproduce

Run the  migrate-to-sw-web-api with docker with account details. 

### Relevant logs and program output

```sh
Unexpected error during fare check: KeyError('greyBoxMessage')
Traceback (most recent call last):
  File "/app/lib/reservation_monitor.py", line 133, in _check_flight_fares
    fare_checker.check_flight_price(flight)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 31, in check_flight_price
    flight_price = self._get_flight_price(flight)
  File "/app/lib/fare_checker.py", line 46, in _get_flight_price
    flights, fare_type = self._get_matching_flights(flight)
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 60, in _get_matching_flights
    change_flight_page, fare_type_bounds = self._get_change_flight_page(flight.reservation_info)
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 92, in _get_change_flight_page
    self._check_for_companion(reservation_info)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 138, in _check_for_companion
    grey_box_message = reservation_info["greyBoxMessage"]
                       ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'greyBoxMessage'
Unexpected error during fare check: KeyError('greyBoxMessage')
Traceback (most recent call last):
  File "/app/lib/reservation_monitor.py", line 133, in _check_flight_fares
    fare_checker.check_flight_price(flight)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 31, in check_flight_price
    flight_price = self._get_flight_price(flight)
  File "/app/lib/fare_checker.py", line 46, in _get_flight_price
    flights, fare_type = self._get_matching_flights(flight)
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 60, in _get_matching_flights
    change_flight_page, fare_type_bounds = self._get_change_flight_page(flight.reservation_info)
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 92, in _get_change_flight_page
    self._check_for_companion(reservation_info)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 138, in _check_for_companion
    grey_box_message = reservation_info["greyBoxMessage"]
                       ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'greyBoxMessage'
Successfully logged in to Thien Huynh's account


Unexpected error during fare check: KeyError('greyBoxMessage')
Traceback (most recent call last):
  File "/app/lib/reservation_monitor.py", line 133, in _check_flight_fares
    fare_checker.check_flight_price(flight)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 31, in check_flight_price
    flight_price = self._get_flight_price(flight)
  File "/app/lib/fare_checker.py", line 46, in _get_flight_price
    flights, fare_type = self._get_matching_flights(flight)
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 60, in _get_matching_flights
    change_flight_page, fare_type_bounds = self._get_change_flight_page(flight.reservation_info)
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 92, in _get_change_flight_page
    self._check_for_companion(reservation_info)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 138, in _check_for_companion
    grey_box_message = reservation_info["greyBoxMessage"]
                       ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'greyBoxMessage'
Unexpected error during fare check: KeyError('greyBoxMessage')
Traceback (most recent call last):
  File "/app/lib/reservation_monitor.py", line 133, in _check_flight_fares
    fare_checker.check_flight_price(flight)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 31, in check_flight_price
    flight_price = self._get_flight_price(flight)
  File "/app/lib/fare_checker.py", line 46, in _get_flight_price
    flights, fare_type = self._get_matching_flights(flight)
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/app/lib/fare_checker.py", line 60, in _get_matching_flights
    change_flight_page, fare_type_bounds = self._get_change_flight_page(flight.reservation_info)
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 92, in _get_change_flight_page
    self._check_for_companion(reservation_info)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 138, in _check_for_companion
    grey_box_message = reservation_info["greyBoxMessage"]
                       ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'greyBoxMessage'
```

**Comment by jdholtz** [comment]
#380 is still a draft so there are bugs in it that are expected for now (such as this one). It is not meant to be used by users at the moment. I may end up ditching #380 entirely, but for now the latest `develop` commit should work with reservations (not accounts yet).

**Comment by donnlee** [comment]
I have a check-in in 3 hours 😬 
So we should use the `develop` branch and `python3 southwest.py CONFIRMATION_NUMBER FIRST_NAME LAST_NAME` ?

**Comment by jdholtz** [comment]
>  So we should use the develop branch and python3 southwest.py CONFIRMATION_NUMBER FIRST_NAME LAST_NAME ?

Yes

**Comment by k3it** [comment]
[develop ](https://github.com/jdholtz/auto-southwest-check-in/tree/develop )branch worked for me with no issues (reservation + name).  thank you @jdholtz for your work.

**Comment by wpkato** [comment]
The develop branch worked for my check-in, (Reservation num+Name).  Thx @jdholtz

**Comment by jdholtz** [comment]
The PR that this issue reports a bug in has been closed (#380). You can see my reasoning on why I don't want to fully switch to use the SW web API [in the comments](https://github.com/jdholtz/auto-southwest-check-in/pull/380#issuecomment-3625266283). In the latest `develop` branch, the script should be fully working again. Please let me know in another issue if you run into problems with the newest `develop` changes.
